Formula/gdal2-python.rb

```diff
@@ -57,23 +57,23 @@ def self.gdal2_py3_exist?
 
   keg_only "older version of gdal is in main tap and installs similar components"
 
-  option "without-python", "Build without Python2 support"
-  option "without-python3", "Build without Python3 support"
+  option "without-python@2", "Build without Python2 support"
+  option "without-python", "Build without Python3 support"
 
   depends_on "swig" => :build
   depends_on "gdal2"
   depends_on NoGdal2Python
+  depends_on "python@2" => :recommended
   depends_on "python" => :recommended
-  depends_on "python3" => :recommended
   depends_on "numpy"
 
   resource "autotest" do
-    url "http://download.osgeo.org/gdal/2.2.3/gdalautotest-2.2.3.tar.gz"
-    sha256 "acc1ec2c8c51830a41557b6c1605f027666a0e2a78898fb9590e75018f57dea6"
+    url "http://download.osgeo.org/gdal/2.2.4/gdalautotest-2.2.4.tar.gz"
+    sha256 "0d3bb1362f86507b255d8e2000b58824b32e9602af893135de549262833bd6aa"
   end
 
   def install
-    if build.without?("python") && build.without?("python3")
+    if build.without?("python@2") && build.without?("python")
```

Comment:
@nickrobison do you know if we should handle this like that here?
see also on line 60-61.
cheers

Reply:
I think this is the new new, correct, way of handling python options, event though it's a bit ugly. I think if we know that we need a version of python for the install to work, then we should fail as early as we can if the constrain isn't satisfied.

Another option would be to use a requirements file, but this seems sufficient for now.